projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,549 @@
+          double stopTimeArrival = firstStop.arrivalTime;
+
+          // Find nearest stop
+          for (StopsAround stopAround : stopsAroundList) {
+
+            ac = Math.abs(i - stopAround.i); // y distance
+            cb = Math.abs(j - stopAround.j); // x distance
+
+            if (distance > Math.hypot(ac, cb)) {
+              distance = Math.hypot(ac, cb);
+              stopTimeArrival = stopAround.arrivalTime;
+            }
+          }
+          // *60 minutes to seconds
+          heatMap[i][j] = (int) ((distance * 60) + stopTimeArrival);
+        }
+        if (heatMap[i][j] == 0) {
+          System.out.println("Check heatMap assignment");
+        }
+      }
+    }
+  }
+
+  // Draw only the heat map colors
+  public void drawHeatMap(String imagePath) {
+    // color1 to color2
+    Color color1 = new Color(255, 208, 235);
+    Color color2 = new Color(126, 0, 67);
+
+    // Normalize data
+    double maxData = 0;
+    double minData = heatMap[10][10];
+    for (int i = 0; i < heatMap.length; i++) {
+      for (int j = 0; j < heatMap[0].length; j++) {
+        if (maxData < heatMap[i][j]) {
+          maxData = heatMap[i][j];
+        }
+        if (heatMap[i][j] != 0 && minData > heatMap[i][j]) {
+          minData = heatMap[i][j];
+        }
+      }
+    }
+
+    // Create image
+    final BufferedImage image =
+        new BufferedImage(heatMap.length, heatMap[0].length, BufferedImage.TYPE_INT_ARGB);
+    final Graphics2D graphics2D = image.createGraphics();
+
+    // Draw the map
+    for (int i = 0; i < heatMap.length; i++) {
+      for (int j = 0; j < heatMap[0].length; j++) {
+        if (heatMap[i][j] == 0) {
+          System.out.println("ERROR:" + "\n" + "heatmap matrix value should not be 0");
+        }
+
+        float value = (float) (heatMap[i][j] - minData) / (float) (maxData - minData);
+
+        int r = (int) (((float) color2.getRed() - color1.getRed()) * value + color1.getRed());
+        int g = (int) (((float) color2.getGreen() - color1.getGreen()) * value + color1.getGreen());
+        int b = (int) (((float) color2.getBlue() - color1.getBlue()) * value + color1.getBlue());
+
+        Color color = new Color(r, g, b);
+        graphics2D.setPaint(color);
+        graphics2D.fillRect(i, j, 2, 2);
+      }
+    }
+    // Draw "X" on stops location
+    for (StopsAround stopAround : stopsAroundList) {
+      graphics2D.setPaint(Color.black);
+      graphics2D.setFont(new Font("TimesRoman", Font.BOLD, 20));
+      graphics2D.drawString("X", stopAround.i - 1, stopAround.j - 1);
+    }
+
+    graphics2D.dispose();
+    // Save file
+    boolean createImage = false;
+    try {
+      createImage = ImageIO.write(image, "png", new File(heatMapPath.get()));
+
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    if (!createImage) {
+      System.out.println("can't create image");
+    }
+  }
+
+  // Functions by different data and sites to get x pixel on the map:
```

Comment:
`Functions by different data and sites to get x pixel on the map` - that sentence is not clear to me. Can you clarify?